src/runtime_musl_workaround.jl

```diff
@@ -0,0 +1,217 @@
+## can do some arithmetic for us, instead of us needing to do manual offset
+## calculation ourselves, which is more error-prone.
+
+
+# Define ELF program header structure, depending on our bitwidth
+@static if Sys.WORD_SIZE == 32
+    struct Elf_Phdr
+        p_type::UInt32
+        p_offset::UInt32
+        p_vaddr::UInt32
+        p_paddr::UInt32
+        p_filesz::UInt32
+        p_memsz::UInt32
+        p_flags::UInt32
+        p_align::UInt32
+    end
+    struct ELF_DynEntry
+        d_tag::UInt32
+        # We drop the `d_un` union, and use only `d_val`, omitting `d_ptr`.
+        d_val::UInt32
+    end
+    else
+    struct Elf_Phdr
+        p_type::UInt32
+        p_flags::UInt32
+        p_offset::UInt64
+        p_vaddr::UInt64
+        p_paddr::UInt64
+        p_filesz::UInt64
+        p_memsz::UInt64
+        p_align::UInt64
+    end
+    struct ELF_DynEntry
+        d_tag::UInt64
+        # We drop the `d_un` union, and use only `d_val`, omitting `d_ptr`.
+        d_val::UInt64
+    end
+end
+
+# Taken from `include/elf.h`
+# https://github.com/ifduyue/musl/blob/aad50fcd791e009961621ddfbe3d4c245fd689a3/include/elf.h#L595
+const PT_DYNAMIC = 2
+# Taken from `include/elf.h`
+#https://github.com/ifduyue/musl/blob/aad50fcd791e009961621ddfbe3d4c245fd689a3/include/elf.h#L735
+const DT_SONAME = 14
+const DT_STRTAB = 5
+
+# This structure taken from `libc.h`
+# https://github.com/ifduyue/musl/blob/aad50fcd791e009961621ddfbe3d4c245fd689a3/src/internal/libc.h#L14-L18
+struct musl_tls_module
+    next::Ptr{musl_tls_module}
+    image::Ptr{musl_tls_module}
+    len::Csize_t
+    size::Csize_t
+    align::Csize_t
+    offset::Csize_t
+end
+
+abstract type musl_dso end
+include(joinpath(@__DIR__, "musl_abi/dso_v1.2.2.jl"))
+include(joinpath(@__DIR__, "musl_abi/dso_v1.1.24.jl"))
+include(joinpath(@__DIR__, "musl_abi/dso_v1.1.22.jl"))
+include(joinpath(@__DIR__, "musl_abi/dso_v1.1.17.jl"))
+include(joinpath(@__DIR__, "musl_abi/dso_v1.1.13.jl"))
+include(joinpath(@__DIR__, "musl_abi/dso_v1.1.12.jl"))
+include(joinpath(@__DIR__, "musl_abi/dso_v1.1.9.jl"))
+include(joinpath(@__DIR__, "musl_abi/dso_v1.1.3.jl"))
+
+function get_musl_dso_type(musl_version::VersionNumber)
+    if musl_version >= v"1.2.2"
+        return musl_dso_v1_2_2
+    elseif musl_version >= v"1.1.24"
+        return musl_dso_v1_1_24
+    elseif musl_version >= v"1.1.22"
+        return musl_dso_v1_1_22
+    elseif musl_version >= v"1.1.17"
+        return musl_dso_v1_1_17
+    elseif musl_version >= v"1.1.13"
+        return musl_dso_v1_1_13
+    elseif musl_version >= v"1.1.12"
+        return musl_dso_v1_1_12
+    elseif musl_version >= v"1.1.10"
+        # I guess v1.1.9's changes didn't stick.  :P
+        return musl_dso_v1_1_3
+    elseif musl_version >= v"1.1.9"
+        return musl_dso_v1_1_9
+    elseif musl_version >= v"1.1.3"
+        return musl_dso_v1_1_3
+    else
+        return nothing
+    end
+end
+
+function get_musl_version()
```

Comment:
Should this (and `get_musl_dso_type`) be a compile-time constant?  How likely is to change the musl version after compilation?

Reply:
I'm concerned that someone will use PackageCompiler on one version of Alpine, then try to run it on another.